lib/spack/spack/__init__.py

```diff
@@ -157,23 +157,22 @@
 #       for packages.
 #
 #-----------------------------------------------------------------------------
-__all__ = ['PackageBase',
-           'Package',
-           'CMakePackage',
-           'AutotoolsPackage',
-           'MakefilePackage',
-           'Version',
-           'when',
-           'ver',
-           'alldeps',
-           'nolink']
-from spack.package import Package, PackageBase, ExtensionConflictError
+__all__ = []
+
+from spack.package import Package
 from spack.build_systems.makefile import MakefilePackage
 from spack.build_systems.autotools import AutotoolsPackage
 from spack.build_systems.cmake import CMakePackage
+__all__ += ['Package', 'CMakePackage', 'AutotoolsPackage', 'MakefilePackage']
```

Comment:
Side comment: how do you feel about a `decorator` to be used in the core to add names to `spack.__all__` ? I was thinking of something like:
```python
@export
class CMakePackage:
    ....
```
If you like the idea I'll prepare a small PR for that.